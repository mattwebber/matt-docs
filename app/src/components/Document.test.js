import React from 'react';
import renderer from 'react-test-renderer';
import { Document } from './';
import { DocumentStore } from '../stores';

describe('Document component', () => {
    let documentStore;
    let stores;

    beforeEach(() => {
        documentStore = new DocumentStore();
        stores = { documentStore };
    });

    it('It should pass document body changes to the store and re-render appropriately', () => {
        const component = renderer.create(
            <Document stores={stores} />
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        const root = component.root;
        const documentInput = root.find(element => element.props.className === 'document-input');
        expect(stores.documentStore.documentBody).toBe('');
        documentInput.props.onChange({ target: { value: 'hello' }});
        expect(stores.documentStore.documentBody).toBe('hello');

        tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});