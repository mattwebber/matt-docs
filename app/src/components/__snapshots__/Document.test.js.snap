// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Document component It should pass document body changes to the store and re-render appropriately 1`] = `
<div
  className="sc-bdfBwQ gEaXWr"
>
  <div
    className="sc-gsTCUz hHmTEf"
  >
    My Document!
  </div>
  <textarea
    className="sc-dlfnbm hHsrQj document-input"
    onChange={[Function]}
    placeholder="Document body..."
    value=""
  />
</div>
`;

exports[`Document component It should pass document body changes to the store and re-render appropriately 2`] = `
<div
  className="sc-bdfBwQ gEaXWr"
>
  <div
    className="sc-gsTCUz hHmTEf"
  >
    My Document!
  </div>
  <textarea
    className="sc-dlfnbm hHsrQj document-input"
    onChange={[Function]}
    placeholder="Document body..."
    value="hello"
  />
</div>
`;

exports[`Document component It should render document body changes 1`] = `
<div
  className="sc-bdfBwQ gEaXWr"
>
  <div
    className="sc-gsTCUz hHmTEf"
  >
    My Document!
  </div>
  <textarea
    className="sc-dlfnbm hHsrQj document-input"
    onChange={[Function]}
    placeholder="Document body..."
    value=""
  />
</div>
`;

exports[`Document component It should render document body changes 2`] = `
<div
  className="sc-bdfBwQ gEaXWr"
>
  <div
    className="sc-gsTCUz hHmTEf"
  >
    My Document!
  </div>
  <textarea
    className="sc-dlfnbm hHsrQj document-input"
    onChange={[Function]}
    placeholder="Document body..."
    value="hello"
  />
</div>
`;
